draft-ietf-jsonpath-base.md

```diff
@@ -1443,17 +1443,39 @@ function-argument   = literal /
                       function-expr
 ~~~
 
-Any function expressions in a query must be well formed (by conforming to the above ABNF)
-and well typed,
+Any function expressions in a query must be well formed (by conforming to the above ABNF),
+supported, and well typed,
 otherwise the JSONPath implementation MUST raise an error
 (see {{synsem-overview}}).
+
+For a function expression to be supported, each function extension involved in the expression
+must be supported.
+Implementations MUST support the function extensions defined
+in this document, but are free to support or not support other registered function extensions.
+
+Implementations MUST NOT support function extensions that are not registered.
+Note: if an implementation supports a function extension that is not registered,
+the implementation will not conform to this specification until the function extension is
+registered.
+To support a new function extension in an implementation, it is recommended that
+the implementation behaves by default in a "strict" mode that conforms to this specification and
+provides some way of behaving in a "non-strict" mode that supports the new function extension.
+This ensures that the default behaviour of the implementation does not mislead users into
+thinking that the function extension is interoperable.
+To use the function extension successfully, users have to consciously choose to use non-strict mode.
+If the function extension is later registered, the implementation can then support the
+function extension in the default, strict mode.
```

Comment:
I don't believe this is necessary.

Reply:
Perhaps not, but I was trying to explain the recommended route to introducing a new function extension. If we omit this sentence, then it gives the impression users will always have to use non-strict mode to use the function extension.

Reply:
Ok, I've deleted that sentence.